pennylane/ops/qubit/matrix_ops.py

```diff
@@ -515,22 +515,27 @@ class BlockEncode(Operation):
     """Gradient computation method."""
 
     def __init__(self, A, wires, id=None):
+        wires = Wires(wires)
         shape_a = qml.math.shape(A)
         if shape_a == () or all(x == 1 for x in shape_a):
             A = qml.math.reshape(A, [1, 1])
-
-        wires = Wires(wires)
-        if pnp.sum(shape_a) <= 2:
-            normalization = A if A > 1 else 1
+            normalization = qml.math.abs(A)
             subspace = (1, 1, 2 ** len(wires))
+
+        elif len(shape_a) == 1:
+            A = qml.math.reshape(A, [1, len(A)])
+            normalization = qml.math.max(A)
+            subspace = (1, shape_a[0], 2 ** len(wires))
+
         else:
-            normalization = max(
+            normalization = qml.math.maximum(
                 norm(A @ qml.math.transpose(qml.math.conj(A)), ord=pnp.inf),
                 norm(qml.math.transpose(qml.math.conj(A)) @ A, ord=pnp.inf),
```

Comment:
Why separate these two cases? We should be able to treat the vector case as a matrix itself, the normalization defined below [ max(norm(A@A^), norm(A^@A)) ]

Reply:
This is @DSGuala design, I just made it traceable 😅

Reply:
The normalization below can't handle a simple list, e.g. `[0.1,0.2]` because ``norm`` doesn't like being called on a scalar. In the next commit, I changed things around so we're reshaping these one dimensional lists into matrices inside the else statement.